lib/sda/host_msg.go

```diff
@@ -56,6 +63,8 @@ func (t *Token) Id() uuid.UUID {
 
 // Return a new Token contianing a reference to the given TreeNode
 func (t *Token) ChangeTreeNodeID(newid uuid.UUID) *Token {
+	tokenMutex.Lock()
+	defer tokenMutex.Unlock()
```

Comment:
If you want something more lightweight (multiple readers at the same time), you can also go with [RWMutex](https://golang.org/pkg/sync/#RWMutex)